src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -40,7 +44,31 @@
      * София - 900</p>
      */
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
-            throws Exception {
-        return null;
+            throws IllegalDateParametersException {
+        if (stringToDate(dateFrom).compareTo(stringToDate(dateTo)) > 0) {
+            throw new IllegalDateParametersException("Wrong parameters");
+        }
+        String report = "Отчёт за период " + dateFrom + " - " + dateTo;
```

Comment:
the checklist says to use a StringBuilder. didn't you see that?

Reply:
Done